src/main/blackbox/blackbox.c

```diff
@@ -1675,10 +1671,25 @@ void blackboxUpdate(timeUs_t currentTimeUs)
  */
 void blackboxInit(void)
 {
+    blackboxResetIterationTimers();
+
+    // an I-frame is written every 32ms
+    // gyro.targetLooptime is 1000 for 1kHz loop, 500 for 2kHz loop etc, gyro.targetLooptime is rounded for short looptimes
+    if (gyro.targetLooptime == 31) { // rounded from 31.25us
+        blackboxIInterval = 1024;
+    } else if (gyro.targetLooptime == 63) { // rounded from 62.5us
+        blackboxIInterval = 512;
+    } else {
+        blackboxIInterval = (uint16_t)(32 * 1000 / gyro.targetLooptime);
+    }
+    // by default p_denom is 32 and a P-frame is written every 1ms
+    // if p_denom is zero then no P-frames are logged
+    blackboxPInterval = blackboxConfig()->p_denom == 0 ? 0 : blackboxIInterval /  blackboxConfig()->p_denom;
```

Comment:
setting `p_denom` too high (33 for 1kHz loop) will disable P-frames completely.
It is also non-trivial to set logging on every iteration (but fixing above problem will fix this)